common/network-common/src/main/java/org/apache/spark/network/util/TransportConf.java

```diff
@@ -363,4 +363,26 @@ public boolean useOldFetchProtocol() {
     return conf.getBoolean("spark.shuffle.useOldFetchProtocol", false);
   }
 
+  /**
+   * The minimum size of a chunk when dividing a merged shuffle file into multiple chunks during
+   * push-based shuffle.
+   * A merged shuffle file consists of multiple small shuffle blocks. Fetching the
+   * complete merged shuffle file in a single response increases the memory requirements for the
```

Comment:
I'm not sure if you're aware of the feature that Spark can save the shuffle data into the disk when the data is too large to hold in the memory.

Reply:
Are you referring to the configuration `maxRemoteBlockSizeFetchToMem`? 
We are aware that when this configuration is set and if a request is larger than this, the block will be saved to disk. 
With push-based shuffle, data of a remote merged block is always large. If we don't divide it into chunks, the remote merged data will always be written to disk and then read from it again. This adds a lot more time.
Also any failure during fetching an entire merged block will be much more costly. With the approach of dividing a merged block into size-able chunks
- We don't have to write to the disk always so the runtime of jobs are shorter.
- When fetch of a shuffle chunk fails, then we fallback to the original blocks corresponding to the mapIds which are part of this chunk.